internal/repos/mysql_conditional_giveaway_repo.go

```diff
@@ -0,0 +1,167 @@
+	var giveaway ConditionalGiveaway
+	err := repo.mysql.WithContext(ctx).SelectOne(&giveaway, "SELECT id, start_time, end_time, guild_id, info_message_id, level FROM conditional_giveaways WHERE guild_id = ? AND end_time IS NULL", guildId)
+	if err != nil {
+		return ConditionalGiveaway{}, err
+	}
+
+	return giveaway, nil
+}
+
+func (repo *ConditionalGiveawayRepo) GetUnfinishedGiveaways(ctx context.Context) ([]ConditionalGiveaway, error) {
+	var giveaways []ConditionalGiveaway
+	_, err := repo.mysql.WithContext(ctx).Select(&giveaways, "SELECT id, start_time, end_time, guild_id, info_message_id, level FROM conditional_giveaways WHERE end_time IS NULL")
+	if err != nil {
+		return nil, err
+	}
+
+	return giveaways, nil
+}
+
+func (repo *ConditionalGiveawayRepo) GetParticipantsForGiveaway(ctx context.Context, giveawayId int) ([]ConditionalGiveawayParticipant, error) {
+	var participants []ConditionalGiveawayParticipant
+	_, err := repo.mysql.WithContext(ctx).Select(&participants, "SELECT id, conditional_giveaway_id, user_id, user_name, join_time, user_level FROM conditional_participants WHERE conditional_giveaway_id = ?", giveawayId)
+	if err != nil {
+		return nil, err
+	}
+
+	return participants, nil
+}
+
+func (repo *ConditionalGiveawayRepo) CountParticipantsForGiveaway(ctx context.Context, giveawayId int) (int, error) {
+	count, err := repo.mysql.WithContext(ctx).SelectInt("SELECT COUNT(*) FROM conditional_participants WHERE conditional_giveaway_id = ?", giveawayId)
+	if err != nil {
+		return 0, err
+	}
+
+	return int(count), nil
+}
+
+func (repo *ConditionalGiveawayRepo) InsertGiveaway(ctx context.Context, guildId, messageId string, level int) error {
+	giveaway := &ConditionalGiveaway{
+		StartTime:     time.Now(),
+		GuildId:       guildId,
+		InfoMessageId: messageId,
+		Level:         level,
+	}
+	err := repo.mysql.WithContext(ctx).Insert(giveaway)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (repo *ConditionalGiveawayRepo) InsertParticipant(ctx context.Context, giveawayId, level int, userId string, userName string) error {
+	participant := &ConditionalGiveawayParticipant{
+		ConditionalGiveawayId: giveawayId,
+		UserId:                userId,
+		UserName:              userName,
+		JoinTime:              time.Now(),
+		UserLevel:             level,
+	}
+	err := repo.mysql.WithContext(ctx).Insert(participant)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (repo *ConditionalGiveawayRepo) InsertWinner(ctx context.Context, giveawayId int, userId, code string) error {
+	winner := &ConditionalGiveawayWinner{
+		ConditionalGiveawayId: giveawayId,
+		UserId:                userId,
+		Code:                  code,
+	}
+	err := repo.mysql.WithContext(ctx).Insert(winner)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (repo *ConditionalGiveawayRepo) UpdateGiveaway(ctx context.Context, giveaway *ConditionalGiveaway, messageId string) error {
```

Comment:
its named wrongly, it should End/Finish Giveaway (Update means just updating all / majority of fields)